mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -506,10 +513,30 @@ abstract class BaseMvRxViewModel<S : MvRxState>(
                 .disposeOnClear()
         }
 
-        val lifecycleAwareObserver = MvRxLifecycleAwareObserver(
+        @Suppress("UNCHECKED_CAST") val lifecycleAwareObserver = MvRxLifecycleAwareObserver(
             lifecycleOwner,
-            alwaysDeliverLastValueWhenUnlocked = !uniqueOnly,
-            onNext = Consumer<T> { subscriber(it) }
+            deliveryMode = deliveryMode,
+            lastDeliveredValue = if (deliveryMode is UniqueOnly) {
+                if (activeSubscriptions.contains(deliveryMode.subscriptionId)) {
+                    throw IllegalStateException("Subscribing with a duplicate subscription id: ${deliveryMode.subscriptionId}. " +
+                        "If you have multiple unique only subscriptions in a MvRx view that listen to the same")
+                }
+                activeSubscriptions.add(deliveryMode.subscriptionId)
+                lastDeliveredStates[deliveryMode.subscriptionId] as? T
```

Comment:
can we remove the value from the map `lastDeliveredStates` too, or does it need to be stored for later?

Reply:
Good catch! we can delete it now since we enforce subscriptions to be unique